client/src/partials/patient_invoice/registry/modalCreditNote.js

```diff
@@ -0,0 +1,48 @@
+angular.module('bhima.controllers')
+.controller('ModalCreditNoteController', ModalCreditNoteController);
+
+ModalCreditNoteController.$inject = [
+  '$uibModalInstance', 'PatientInvoiceService', 'util', 'data', 'JournalVoucherService', 'VoucherService'
+];
+
+function ModalCreditNoteController( $uibModalInstance, Invoices, Util, data, JournalVoucher, Vouchers) {
+  var vm = this;
+  vm.patientInvoice = data.invoice; 
+  vm.submit = submit;
+  vm.cancel = cancel;
+  vm.creditNote;
+
+  // transfer type
+  vm.transferType = Vouchers.transferType;
+
+  var typeId = vm.transferType.filter(function (item) {
+    return item.incomeExpense === 'creditNote';
+  });
+
+  var transferTypeId = typeId[0].id;
+
+
+  Invoices.read(vm.patientInvoice.uuid)    
+    .then(function (data){
+      vm.patientInvoiceItems = data.items;
+    });
+
+  function submit(uuid, creditNote) {
+    creditNote.type_id = transferTypeId;
+    
+    if (!uuid || !creditNote) { return; }
```

Comment:
Should this give some sort of error message? How will the user know that nothing is happening?